tests/middleware/fieldPermissions.test.js

```diff
@@ -0,0 +1,267 @@
+const tape = require('tape')
+const supertest = require('supertest')
+const server = require('../../src/server')
+
+const boomErrorHandler = require('../../src/middleware/boomErrorHandler')
+const fieldPermissions = require('../../src/middleware/fieldPermissions')
+const validateHeaderToken = require('../../src/middleware/validateHeaderToken')
+const { getUnauthorizedFields } = require('../../src/helpers/permissions')
+
+const User = require('../../src/models/User.js')
+const Event = require('../../src/models/Event.js')
+const Token = require('../../src/models/auth/Token.js')
+
+const { validAdminUser, validBasicUser } = require('../fixtures/users.json')
+const { validEvent2 } = require('../fixtures/events.json')
+const { token } = require('../fixtures/auth/tokens.json')
+
+const adminUserToken = Object.assign({}, token, { accessToken: '456' })
+const basicUserToken = Object.assign({}, token, { accessToken: '123' })
+
+const roles = require('../../src/constants/roles.js')
+
+// 1) prepare the db
+
+tape('emptying db.', t => {
+  Promise.all([
+    User.remove({}),
+    Event.remove({}),
+    Token.remove({})
+  ])
+  .then(() => t.end())
+  .catch(err => t.end(err))
+})
+
+// we will create a basic user, an admin user, an event owned by the admin user,
+// and tokens to sign in either user
+
+let adminUserId, basicUserId, eventId, adminToken, basicToken
+
+tape('filling db.', t => {
+  Promise.all([
+    User.create(validBasicUser),
+    User.create(validAdminUser)
+  ])
+  .then(([ basicUser, adminUser ]) => {
+    basicUserId = basicUser.id
+    adminUserId = adminUser.id
+    validEvent2.owner = adminUserId
+    adminUserToken.user = adminUserId
+    basicUserToken.user = basicUserId
+    return [ validEvent2, adminUserToken, basicUserToken ]
+  })
+  .then(([ validEvent2, adminUserToken, basicUserToken ]) => Promise.all([
+    Event.create(validEvent2),
+    Token.create(adminUserToken),
+    Token.create(basicUserToken)
+  ]))
+  .then(([ event, adminUserToken, basicUserToken ]) => {
+    adminToken = adminUserToken.accessToken
+    basicToken = basicUserToken.accessToken
+    eventId = event.id
+  })
+  .then(() => t.end())
+  .catch(err => t.end(err))
+})
+
+// 2) add dummy routes for testing purposes
+
+const testPermissions = {
+  f1: [ roles.SUPER ],
+  f2: [ roles.ADMIN ],
+  f3: [ roles.BASIC ],
+  f4: [ roles.SUPER, roles.OWNER ],
+  f5: [ roles.ADMIN, roles.OWNER ],
+  f6: [ roles.BASIC, roles.OWNER ]
+}
+
+// remove '/test' from req urls so fieldPermissions can get resource type from it
+const urlSlice = (req, res, next) => {
+  req.url = req.url.slice(5)
+  next()
+}
+
+server.post('/test/events',
+  validateHeaderToken,
+  urlSlice,
+  fieldPermissions(testPermissions),
+  (req, res, next) => res.send('success!'),
+  boomErrorHandler
```

Comment:
Do you need this?

Reply:
Yes, we are testing a fake route and some of the tests are to check we get sent back the correct error